### PR TITLE
🐛 Detect encodings of files when reading

### LINKED ISSUE #508: read_df confused by (useless in utf8) byte order marks
See: https://github.com/kids-first/kf-lib-data-ingest/pull/493#discussion_r497682553
And: https://docs.python.org/3/library/codecs.html#encodings-and-unicode

I think read_df should understand file encoding when reading files

### PR DESCRIPTION
**Pull request by fiendish**
closes #508

### CODE REVIEW

File: kf_lib_data_ingest/common/io.py [hunk 1]
```diff
@@ -62,7 +74,11 @@ def read_delimited_text_df(filepath_or_buffer, **kwargs):
     kwargs["engine"] = "python"
     kwargs["dtype"] = str
     kwargs["na_filter"] = False
-    return pandas.read_csv(__ascii_wrap(filepath_or_buffer), **kwargs)
+    kwargs["encoding"] = kwargs.get("encoding") or __detect_encoding(
```
Comment (znatty22) [inline kf_lib_data_ingest/common/io.py:77]: Hmm is this going to slow things down in a noticeable way for big dfs? Couple options.. 1) Could make the default behavior what we had before (no supplied encoding to read_df) and then add an `auto_detect_encoding` kwarg in read_df or 2) Always assume large text, detect encoding incrementally, stop when you find it: https://chardet.readthedocs.io/en/latest/usage.html?#advanced-usage

**Review by znatty22 — CHANGES_REQUESTED** [review]

File: kf_lib_data_ingest/common/io.py (hunk 1 above)
Comment (fiendish) [reply kf_lib_data_ingest/common/io.py:77]: It shouldn't. It only reads a 2kb from the file.

Comment (znatty22) [reply kf_lib_data_ingest/common/io.py:77]: Ohhh ok I was looking at chardet. I'll approve

**Review by znatty22 — APPROVED** [review]
⚡

**Review by znatty22 — APPROVED** [review]
👍

**Pull request merged**